### ISSUE TITLE
FR: `setDT` for matrices

### ISSUE THREAD

**Issue by iagogv3** [Rdatatable/data.table#6565]
Currently, `setDT` accepts as main argument
```Rd
x: A named or unnamed ‘list’, ‘data.frame’ or ‘data.table’
```
Would it have sense to accept a matrix too instead of asignment `dt <- as.data.table(matrix(...))`?

**Comment by ecoRoland2** [comment]
I don't think that makes sense. A data.frame and a data.table are lists. A matrix is a very different data structure. How often do you create data.tables from matrices that are so large that you need to avoid a deep copy?

**Comment by ChristK** [comment]
@ecoRoland2 Leaving implementation difficulties aside, I often do simulations using Rcpp and matrices, as matrices are much more intuitive to work with on the C++ side. The outputs are also matrices, sometimes several GB each. Having a more efficient way to convert a matrix to a data.table on the R side would be nice. So, the request by @iago-pssjd is not outrageous, and there are real-life use cases, albeit perhaps rare.

**Comment by MichaelChirico** [comment]
I think you'll have to do this yourself, like

```
DT=data.table(M[, 1])
# maybe setalloccol() if M is very wide
for (j in 2:ncol(M))
  set(DT,,j, M[, j])
setnames(DT, colnames(M))
```

this could conceivably be wrapped into a data.table export, though I worry about the troubles of creating a column-by-column by-reference copy of an array.

it will definitely not be part of setDT, the semantics are way too different.

**Comment by ben-schwen** [comment]
AFAIR there is a PR for a fast matrix to data.table conversion.

**Comment by iagogv3** [comment]
Given the comments above, I'll close this for the moment
